### ISSUE TITLE
Allow custom template function or handlebars template syntax

### ISSUE THREAD

**Issue by scopsy** [notifirehq/notifire#31]
- **I'm submitting a ...**
  [ ] bug report
  [x] feature request
  [ ] question about the decisions made in the repository
  [ ] question about how to use this project

- **Summary**
Give the ability to create template content based on user-defined functions and as well as handlebars. 
so changes will be as follows

```ts
interface IMessage {
  subject?: string;
  channel: ChannelTypeEnum;
  template: string | ((paras: any[]) => Promise<string> | string);  // <- this is updated from string
  active?: boolean | ((payload: ITriggerPayload) => Promise<boolean> | boolean);
}
```

- **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

**Comment by davidsoderberg** [comment]
@ComBarnea I have a proposal to solve this enhancement if you like me to do a PR?

**Comment by scopsy** [comment]
Sure @davidsoderberg sounds amazing. Let me know if you need any help 🤟

**Comment by davidsoderberg** [comment]
@scopsy what branch should I create PR against?

**Comment by scopsy** [comment]
@davidsoderberg master is fine

**Comment by scopsy** [comment]
Closed by #85 Thank you so much @davidsoderberg !!
